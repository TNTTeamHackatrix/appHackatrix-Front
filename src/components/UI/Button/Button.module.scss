@import "../../../scss/variables";

.Button {
  border: 1px solid transparent;
  font-style: normal;
  font-stretch: normal;
  font-weight: bold;
  line-height: normal;
  letter-spacing: normal;
  padding: 0 15px;
  text-align: center;
  text-transform: uppercase;

  &:focus {
    outline: none;
  }
}

.ButtonFullWidth {
  width: 100%;
}

.ButtonNotFullWidth {
  width: auto;
}

.ButtonPrimary {
  background-color: #00aec7;
  border-radius: 4px;
  color: $white;
  font-size: 16px;
  height: 48px;
  margin-left: 5px;
  padding: 0 15px;
  text-align: center;
  text-transform: uppercase;
  cursor: pointer;

  &:disabled {
    background-color: $black;
    opacity: .4;
  }
}

.ButtonPrimaryLoader {
  background-color: #00aec7;
  border-radius: 4px;
  color: $white;
  font-size: 16px;
  height: 48px;
  padding: 0 15px;
  text-align: center;
  text-transform: uppercase;
  cursor: pointer;

  &:disabled {
    background-color: #00aec7;
    opacity: .4;
  }
}

.ButtonSecondary {
  background-color: $sickly-yellow;
  border-radius: 20px;
  color: #0134a1;
  height: 40px;
  font-size: 12px;
  font-weight: bold;
  cursor: pointer;
  text-transform: uppercase;
}

.ButtonOutline {
  background-color: transparent;
  border: 0;
  color: rgba(51, 51, 51, 0.7);
  font-size: 16px;
  height: 48px;
  margin-right: 5px;
  padding: 0 15px;
  text-align: center;
  text-transform: uppercase;
  cursor: pointer;

  &:disabled {
    background-color: $black;
    opacity: .4;
  }
}

.ButtonGhost {
  background-color: transparent;
  border: 1px solid $darkish-blue;
  border-radius: 24px;
  color: $darkish-blue;
  height: 40px;
  font-size: 16px;
  font-weight: bold;
  text-transform: uppercase;
  cursor: pointer;

  &:disabled {
    border: 1px solid $black;
    color: $black;
    opacity: .4;
  }
}

@media (max-width: 450px) {
  .Button {
    margin: 0;
  }
}
