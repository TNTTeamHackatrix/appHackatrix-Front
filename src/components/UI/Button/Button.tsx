import React from "react";
import styles from "./Button.module.scss";
import classNames from "classnames";
import AnimationLoader from "../AnimationLoader/AnimationLoader";

export type ButtonProps = {
  id?: string;
  cssClassName?: string;
  disabled?: boolean;
  loader?: boolean;
  type: string;
  name: string;
  onClick?: React.MouseEventHandler<any>;
  typeStyle?: "primary" | "secondary" | "outline" | "ghost";
  notFullWidth?: boolean;
};

class Button extends React.Component<ButtonProps> {

  render() {

    let width;
    if (this.props.notFullWidth === true) {
      width = "ButtonNotFullWidth";
    } else {
      width = "ButtonFullWidth";
    }

    let buttonTypeStyle = "ButtonPrimary";
    if (this.props.typeStyle === "secondary") {
      buttonTypeStyle = "ButtonSecondary";
    } else if (this.props.typeStyle === "outline") {
      buttonTypeStyle = "ButtonOutline";
    } else if (this.props.typeStyle === "ghost") {
      buttonTypeStyle = "ButtonGhost";
    }

    let button;

    if (!this.props.loader) {
      button = (
        <button
          id={this.props.id}
          name={this.props.name}
          type={this.props.type}
          className={classNames(styles.Button, this.props.cssClassName, styles[buttonTypeStyle], styles[width])}
          disabled={this.props.disabled}
          onClick={this.props.onClick}
        >
          {this.props.children}
        </button>
      );
    } else {
      button = (
        <button
          id={this.props.id}
          name={this.props.name}
          type={this.props.type}
          className={classNames(styles.Button, styles.ButtonPrimaryLoader, styles[buttonTypeStyle], styles[width])}
          disabled={this.props.disabled}
          onClick={this.props.onClick}
        >
          <AnimationLoader/>
        </button>
      );
    }

    return (
      <div>
        {button}
      </div>
    );
  }
}

export default Button;
