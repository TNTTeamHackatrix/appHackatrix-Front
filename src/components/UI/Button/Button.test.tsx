import React from "react";
import {mount, shallow} from "enzyme";
import renderer from "react-test-renderer";
import Button from "./Button";

describe("Button", () => {

  const setup = propOverrides => {
    return Object.assign({
      name: undefined,
      type: undefined,
      disabled: undefined,
      typeStyle: undefined,
      notFullWidth: undefined,
      loader: undefined
    }, propOverrides);
  };

  it("always renders a button tag", () => {
    const ButtonComponent = renderer.create(<Button {...setup({})}/>).toJSON();
    expect(ButtonComponent).toMatchSnapshot();
  });

  it("should be disabled when prop 'disabled' is true", () => {
    const props = setup({ disabled: true });
    const ButtonComponent = shallow(<Button {...props}/>);
    expect(ButtonComponent.find("button").prop("disabled")).toEqual(true);
  });

  test.each`
    typeStyle | className
    ${"primary"} | ${"ButtonPrimary"}
    ${"secondary"} | ${"ButtonSecondary"}
    ${"outline"} | ${"ButtonOutline"}
    ${"ghost"} | ${"ButtonGhost"}
  `("should have class $className when prop 'typeStyle' is $typeStyle", ({typeStyle, className}) => {
    const props = setup({ typeStyle: typeStyle });
    const ButtonComponent = shallow(<Button {...props}/>);
    expect(ButtonComponent.find("button").hasClass(className)).toEqual(true);
  });

  test.each`
    width | className
    ${true} | ${"ButtonNotFullWidth"}
    ${false} | ${"ButtonFullWidth"}
  `("should have class $className when prop 'notFullWidth' is $width", ({width, className}) => {
    const props = setup({ notFullWidth: width });
    const ButtonComponent = shallow(<Button {...props}/>);
    expect(ButtonComponent.find("button").hasClass(className)).toEqual(true);
  });

  it("should render $AnimationLoader when prop 'loader' is true", () => {
    const props = setup({ loader: true });
    const ButtonComponent = mount(<Button {...props}/>);
    expect(ButtonComponent.find("button").children()).toHaveLength(1);
  });

  it("should render a text content when there's any text inside de tag", () => {
    const ButtonComponent = mount(<Button {...setup({})}>Test</Button>);
    expect(ButtonComponent.find("button").text()).toBe("Test");
  });

  it("should execute a function when onClick", () => {
    const mockCallBack = jest.fn();
    const ButtonComponent = shallow((<Button {...setup({})} onClick={mockCallBack}/>));
    ButtonComponent.find("button").simulate("click");
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });

});
