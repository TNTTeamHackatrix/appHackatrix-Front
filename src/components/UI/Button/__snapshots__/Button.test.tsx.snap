// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button always renders a button tag 1`] = `
<div>
  <button
    className="Button ButtonPrimary ButtonFullWidth"
  />
</div>
`;
